modules/host/contractmanager/writeaheadlog.go

```diff
@@ -225,22 +450,24 @@ func (wal *writeAheadLog) recoverWAL(walFile file) error {
 	// changes can be parsed properly.
 	var sc stateChange
 	var scs []stateChange
-	for err == nil {
-		err = decoder.Decode(&sc)
-		if err == nil {
-			// The uncommitted changes are loaded into memory using a simple
-			// append, because the tmp WAL file has not been created yet, and
-			// will not be created until the sync loop is spawned. The sync
-			// loop spawner will make sure that the uncommitted changes are
-			// written to the tmp WAL file.
-			wal.commitChange(sc)
-			scs = append(scs, sc)
+	offset := headerLength() + header.LengthMD
+	for offset < maxWalSize {
```

Comment:
This loop might need to be changed to run until an error occured. There is a slim chance that the reset which is initialized at 80% of `maxWalSize` doesn't finish in time. That means that the WAL grows larger than `maxWalSize` and is cut off in this loop. 
Of course other checks that use maxWalSize would have to be modified accordingly since maxWalSize wouldn't be the actual maxWalSize anymore but rather a soft limit.